kedro/io/kedro_data_catalog.py

```diff
@@ -231,30 +273,31 @@ def _get_dataset(
     def add(
```

Comment:
Can (should?) we add a method to remove a dataset? Either ``pop()`` or ``del`` (I saw elsewhere you did not ant to maintain all dict like methods, but I think this may be useful). I think it could be useful when one copy a catalog, filter a pipeline and just want to get some datasets associated to the filtered pipeline.

Reply:
We didn't plan to add them in the first version, but we can consider doing so if we see user demand. The main reason for not adding them now is that we still don't want to make catalog fully mutable.

Reply:
Yes, I understand that we want to avoid users to modify the catalog inplace in a hook. As a user I'd assume I could handle it like a dict, but this deleting a key prgramatically seems a very rare use case.